### PR TITLE
prov/rxm: code cleanups

### PR DESCRIPTION
**Pull request by nikitaxgusev**
Signed-off-by: Nikita Gusev <nikita.gusev@intel.com>

### CODE REVIEW

File: prov/rxm/src/rxm_conn.c [hunk 1]
```diff
@@ -1348,10 +1347,7 @@ static void *rxm_conn_atomic_progress(void *arg)
 
 	FI_INFO(&rxm_prov, FI_LOG_EP_CTRL, "Starting auto-progress thread\n");
 	while (ep->do_progress) {
-		/* TODO: Remove this lock, it can't protect anything */
-		ofi_ep_lock_acquire(&ep->util_ep);
 		ret = fi_trywait(fabric->msg_fabric, fids, 2);
-		ofi_ep_lock_release(&ep->util_ep);
```
Comment (shefty) [inline prov/rxm/src/rxm_conn.c]: Please separate this into its own patch.

Comment (nikitaxgusev) [reply prov/rxm/src/rxm_conn.c]: @shefty Added a new commit which contains the removing an unnecessary lock.

**Comment by nikitaxgusev** [comment]
@shefty done.

**Comment by nikitaxgusev** [comment]
@shefty let's rerun testing. And merge it, if it is ok. Thanks.

**Comment by shefty** [comment]
Can you please rebase against the tip of master?  CI is failing, but I'm pretty sure it's unrelated to the changes.

**Comment by rajachan** [comment]
add to whitelist
bot:aws:retest

**Comment by nikitaxgusev** [comment]
@shefty sorry for delay, rebased.
